fdmprinting/archive_retrieve.py

```diff
@@ -153,20 +158,36 @@ def get_status_from_mqtt(printer, printer_id):
             tls=tls
         )
         printer_object = json.loads(msg.payload)
-
+     
         status["name"] = printer["name"]
         status["printer_id"] = printer_id
         status["state"] = printer_object["print"]["gcode_state"]
         status["job"] = printer_object["print"]["subtask_name"]
         status["mins"] = printer_object["print"]["mc_remaining_time"]
         status["task_id"] = printer_object["print"]["task_id"]
         status["raw_json"] = str(printer_object)
+        status["fail_reason"] = printer_object["print"]["fail_reason"]
     except Exception as e:
         print(f'Failed getting status for {printer["name"]} ({printer["ip"]}:{printer["port"]})', e)
 
     return status
 
 
+def get_status_msg(status):
+    if status["fail_reason"] != 0:
+       
+       # Convert and addreses the fixed width format 
+       # of const_print_errors.py
+       hex_code = f'{status["fail_reason"]:x}'
+       if len(hex_code) == 7: hex_code = "0" + str(hex_code)
+       
+       for key, value in PRINT_ERROR_ERRORS.items():
+            if hex_code.upper() == key:
+               return "\n\n**" + value + "**\n\n"
+    
+    return "\n\n"
+
+
```

Comment:
Please check your IDE to ensure consistent settings: not at all a blocker, but to date we've done 4 spaced (`    `) and you're mixing 3 and 5 here.

```suggestion
def get_status_msg(status):
    if status["fail_reason"] != 0:
       
        # Convert and addreses the fixed width format
        # of const_print_errors.py
        hex_code = f'{status["fail_reason"]:x}'
        if len(hex_code) == 7: hex_code = "0" + str(hex_code)
    
        for key, value in PRINT_ERROR_ERRORS.items():
            if hex_code.upper() == key:
                return "\n\n**" + value + "**\n\n"
    
    return "\n\n"


```